### DISCUSSION TITLE
Can't connect Sonoff Basic to Home app

### DISCUSSION THREAD

**Discussion by kossinow** [RavenSystem/esp-homekit-devices#1978] [Q&A]
Hello, nice fellas!

I have Sonoff Basic on esp8285
Windows 11 21H2 (Python 3.10.10, esptool 4.5)
iPhone 13 (iOs 16.3.1)

I do:
1. get fullhaaboot.bin SHA256 (182eb2af7cfbffe6f7c634c3e77a65ae5108de71d535a74db935fffbbd8dd0ec)
2. perform:
python3 -m esptool erase_flash
python3 -m esptool -b 115200 write_flash -fs 1MB -fm dout -ff 40m 0x0 fullhaaboot.bin
3. connect SSID: HAA-4B8CB7 and open 192.168.4.1:4567 (Safari)
4. Fill in parameters in HAA installer 6.6.1 page:
	MELPHAA Script: {"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0}]}]}
	WiFI Mode: Normal
	Flash 14325e DOUT: DOUT
	Choose my WiFi and put password
	Update Server: empty
	Port: 80
	HTTPS: v
	hit Save and get OK page

	(also tried put github.com/RavenSystem/haa/releases/download/6.6.1 with port 443)
5. wait about 5 mins and see SSID: Link_A868 periodically
6. try to scan QR with Home app
7. when it fails reload app and device, try again
8. nothing happens

Can you point what's wrong?

**Comment by xrust83** [comment]
https://www.youtube.com/watch?v=O7lvkPdyyNY 

[Інструкція HAA в HomeKit.pdf](https://github.com/RavenSystem/esp-homekit-devices/files/10870632/HAA.HomeKit.pdf)

**Reply by xrust83** [reply]
Your problem may be related to an error in the MEPLHAA script. Sometimes when copying the code, the symbol {"} is replaced by an incompatible one, for example, in the Urkainese language, these are other symbols according to the character code. The problem is that when you did everything correctly, but an error occurs in MEPLHAA, the device goes into configuration mode, but already in your local network , while it does not create an access point with the IP 192.168.4.1:4567, but simply opens the local configuration page with the IP 192.168.Х.Х:4567, which already belongs to your local network. You need to find out what IP the router has assigned to the device. And go and check the correctness of the MEPLHAA script

**Reply by kossinow** [reply]
Well I see my device is connected to network, but when I try to connect http://192.168.1.240:4567 browser says "refused to connect".
Ando one more, when I scan for devices in Home app it shows my macbook, when I try to connect, get error message which says that iphone probably in another network.
Is it possible that problem is in the router?

**Reply by xrust83** [reply]
the web interface is open for connection for a few minutes after loading, then it hides.  you need to enter the device into setup mode.  To do this, there are many ways to roughly turn it on and off 8-10 times until a more gentle one when you turn on the power and do not let the device boot up completely, turn off the power.  2-3 such actions will force the board to go into setup mode.  after that, a page will appear at the specified address http://192.168.1.240:4567 on the page, check the box next to homekit ID and save the settings.  after that, your device should again request pairing with hommekit

**Reply by kossinow** [reply]
I managed to get to that page and done reset homekit ID.
Still no connection

**Reply by nickforev** [reply]
Hi. Your phone connection to same network 192.168.1.1?

**Reply by kossinow** [reply]
I use my phone's browser and it in the same network as i mentioned in step 4 (HAA), also I turned off mobile data.

**Reply by xrust83** [reply]
If he went to the setup mode page and reset the homekit id, then theoretically the network is the same.  Can you take a video of what you're doing from another phone?

**Reply by kossinow** [reply]
> If he went to the setup mode page and reset the homekit id, then theoretically the network is the same. Can you take a video of what you're doing from another phone?

https://youtu.be/tNvKyuTxDFQ
